core/src/main/scala/org/apache/spark/shuffle/BlockStoreShuffleReader.scala

```diff
@@ -46,7 +46,9 @@ private[spark] class BlockStoreShuffleReader[K, C](
       context,
       blockManager.shuffleClient,
       blockManager,
-      mapOutputTracker.getMapSizesByExecutorId(handle.shuffleId, startPartition, endPartition),
+      mapOutputTracker.getMapSizesByExecutorId(
```

Comment:
seems like we can just pass a `BaseShuffleHandle`, then we can get `shuffleId` and `dep.serializer.supportsRelocationOfSerializedObjects` inside the method.

Reply:
nvm, the current one is easier to test